dashboard/app/assets/javascripts/videos.js.erb

```diff
@@ -23,15 +23,22 @@ function createVideo(options) {
 //   key - an uid.
 //   name - a string.
 //   redirect - the redirect page after the video is dismissed.
+//   onClose - actions to take after closing the video dialog, or immediately
+//             if the video isn't shown.
 function showVideoDialog(options, forceShowVideo) {
   if (forceShowVideo === undefined) {
     forceShowVideo = false;
   }
 
+  if (options.onClose === undefined) {
+    options.onClose = function () {};
+  }
+
   if (dashboard.clientState.hasSeenVideo(options.key) && forceShowVideo === false) {
     // Anything we were going to do when the video closed, we ought to do
     // right now.
     // TODO: What else are we missing here?
```

Comment:
I don't think you are missing anything. This is the only place where we skip showing the video